src/lib.rs

```diff
@@ -311,4 +339,64 @@ mod tests {
         let raw_parts = RawParts::from_vec(vec);
         assert_eq!(raw_parts.capacity, 100);
     }
+
+    #[test]
+    fn debug_test() {
+        let mut vec = Vec::with_capacity(100); // capacity is 100
+        vec.extend_from_slice(b"123456789"); // length is 9
+        let raw_parts = RawParts::from_vec(vec);
+
+        assert_eq!(
+            format!(
+                "RawParts {{ ptr: {:?}, length: 9, capacity: 100 }}",
+                raw_parts.ptr
+            ),
+            format!("{:?}", raw_parts)
+        );
+    }
+
+    #[test]
+    fn partial_eq_fail_pointer() {
+        let mut vec_1 = Vec::with_capacity(100); // capacity is 100
+        vec_1.extend_from_slice(b"123456789"); // length is 9
+        let mut vec_2 = Vec::with_capacity(100); // capacity is 100
+        vec_2.extend_from_slice(b"123456789"); // length is 9
+
+        let raw_parts_1 = RawParts::from_vec(vec_1);
+        let raw_parts_2 = RawParts::from_vec(vec_2);
+        assert_ne!(raw_parts_1, raw_parts_2);
+    }
+
+    #[test]
+    fn partial_eq_fail_capacity() {
+        let mut vec_1 = Vec::with_capacity(100); // capacity is 100
+        vec_1.extend_from_slice(b"123456789"); // length is 9
+        let mut vec_2 = Vec::with_capacity(101); // capacity is 101
+        vec_2.extend_from_slice(b"123456789"); // length is 9
+
+        let raw_parts_1 = RawParts::from_vec(vec_1);
+        let raw_parts_2 = RawParts::from_vec(vec_2);
+        assert_ne!(raw_parts_1, raw_parts_2);
+    }
+
+    #[test]
+    fn partial_eq_fail_length() {
+        let mut vec_1 = Vec::with_capacity(100); // capacity is 100
+        vec_1.extend_from_slice(b"123456789"); // length is 9
+        let mut vec_2 = Vec::with_capacity(100); // capacity is 100
+        vec_2.extend_from_slice(b"12345678"); // length is 8
+
+        let raw_parts_1 = RawParts::from_vec(vec_1);
+        let raw_parts_2 = RawParts::from_vec(vec_2);
+        assert_ne!(raw_parts_1, raw_parts_2);
+    }
+
+    #[test]
+    fn partial_eq_pass() {
+        let mut vec = Vec::with_capacity(100); // capacity is 100
+        vec.extend_from_slice(b"123456789"); // length is 9
+
+        let raw_parts = RawParts::from_vec(vec);
+        assert_eq!(raw_parts, raw_parts);
```

Comment:
We can address this clippy warning with the suggestion below:

```
warning: identical args used in this `assert_eq!` macro call
   --> src/lib.rs:400:20
    |
400 |         assert_eq!(raw_parts, raw_parts);
    |                    ^^^^^^^^^^^^^^^^^^^^
    |
note: the lint level is defined here
   --> src/lib.rs:1:9
    |
1   | #![warn(clippy::all)]
    |         ^^^^^^^^^^^
    = note: `#[warn(clippy::eq_op)]` implied by `#[warn(clippy::all)]`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#eq_op
```

```suggestion
        let RawParts {
            ptr,
            length,
            capacity,
        } = RawParts::from_vec(vec);
        let a = RawParts {
            ptr,
            length,
            capacity,
        };
        let b = RawParts {
            ptr,
            length,
            capacity,
        };
        assert_eq!(a, b);
```

Reply:
Thanks for fixing that, didn't realize it would trigger a warning.